### PR TITLE
Support head operations

### PR DESCRIPTION
**Pull request by hectorcast-db**
## Changes
Support head operations

## Tests
Generated SDK and run Files integration tests.

### CODE REVIEW

**Review by edwardfeng-db — APPROVED** [review]

File: databricks-sdk-java/src/main/java/com/databricks/sdk/support/Header.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.databricks.sdk.support;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Header {
```
Comment (mgyucht) [inline databricks-sdk-java/src/main/java/com/databricks/sdk/support/Header.java:10]: Can you add a doccomment here?

File: databricks-sdk-java/src/main/java/com/databricks/sdk/core/ApiClient.java [hunk 2]
```diff
@@ -319,11 +333,18 @@ private String makeLogRecord(Request in, Response out) {
...
     if (target == InputStream.class) {
-      return (T) body;
+      return (T) response.getBody();
     }
-    return mapper.readValue(body, target);
+    T object;
+    try {
+      object = target.getDeclaredConstructor().newInstance();
+    } catch (Exception e) {
+      throw new IOException("Unable to initialize an instance of type " + target.getName());
```
Comment (mgyucht) [inline databricks-sdk-java/src/main/java/com/databricks/sdk/core/ApiClient.java:344]: We should throw only DatabricksException here

File: databricks-sdk-java/src/test/java/com/databricks/sdk/integration/FilesIT.java [hunk 3]
```diff
@@ -44,7 +44,10 @@ private void writeFileAndReadFileInner(
 
     // Write the file to DBFS.
     workspace.files().upload(fileName, inputStream);
-
+    // TODO: Enable after SDK generated
```
Comment (mgyucht) [inline databricks-sdk-java/src/test/java/com/databricks/sdk/integration/FilesIT.java:47]: Reminder to uncomment this

**Review by mgyucht — COMMENTED** [review]
Overall, looking good. Let's make sure there is at least 1 integration test to cover this case before we merge, if possible.

**Review by mgyucht — APPROVED** [review]
Approved (though we'll test in a follow-up PR).

**Pull request merged**